spec/core/ClockSpec.js

```diff
@@ -687,6 +687,136 @@ describe('Clock (acceptance)', function() {
     expect(recurring1.calls.count()).toBe(4);
   });
 
+  describe('autoTick mode', () => {
+    let delayedFunctionScheduler;
+    let mockDate;
+    let clock;
+    const realTimeout = setTimeout;
+
+    beforeEach(() => {
+      delayedFunctionScheduler = new jasmineUnderTest.DelayedFunctionScheduler();
+      mockDate = {
+        install: function() {},
+        tick: function() {},
+        uninstall: function() {}
+      };
+      clock = new jasmineUnderTest.Clock(
+        // We use the real window for global or firefox is displeased when we try to call a real setTimeout on an object "that doesn't implement window".
+        typeof window !== 'undefined' ? window : { setTimeout: setTimeout },
+        function() {
+          return delayedFunctionScheduler;
+        },
+        mockDate
+      );
+      clock.install();
+    });
+
+    afterEach(() => {
+      clock.uninstall();
+    });
+
+    it('can run setTimeouts/setIntervals asynchronously', function() {
+      const recurring = jasmine.createSpy('recurring'),
+        fn1 = jasmine.createSpy('fn1'),
+        fn2 = jasmine.createSpy('fn2'),
+        fn3 = jasmine.createSpy('fn3');
+
+      const intervalId = clock.setInterval(recurring, 50);
+      // In a microtask, add some timeouts.
+      Promise.resolve()
+        .then(function() {
+          return new Promise(function(resolve) {
+            clock.setTimeout(resolve, 25);
+          });
+        })
+        .then(function() {
+          fn1();
+          return new Promise(function(resolve) {
+            clock.setTimeout(resolve, 200);
+          });
+        })
+        .then(function() {
+          fn2();
+          return new Promise(function(resolve) {
+            clock.setTimeout(resolve, 100);
+          });
+        })
+        .then(function() {
+          fn3();
+        });
+
+      expect(recurring).not.toHaveBeenCalled();
+      expect(fn1).not.toHaveBeenCalled();
+      expect(fn2).not.toHaveBeenCalled();
+      expect(fn3).not.toHaveBeenCalled();
+
+      clock.setAutoTickMode(true);
```

Comment:
For the sake of providing a stable yet evolvable API, I would really recommend making this function take an enum instead of Boolean. It's hard to know if some future subtle change will be needed to the tick mode. Making this multiple choice rather than true false can allow for experimentation and smooth upgrades.

Reply:
I'm certainly open to making that change. I don't know of any enum-like inputs in jasmine today so this might be a bike-shedding question for the maintainers for how best to do this.

// cc @sgravrock

Reply:
I've updated the implementation to use a string union. Since this is only used for tests (no bundle size concerns), I felt that'd be more ergonomic than an enum/object alternative.